src/nupic/types/Sdr.hpp

```diff
@@ -27,17 +27,21 @@
 #include <algorithm> //sort
 #include <functional>
 #include <vector>
+
 #include <nupic/types/Types.hpp>
 #include <nupic/types/Serializable.hpp>
 #include <nupic/utils/Random.hpp>
 
 namespace nupic {
 namespace sdr {
 
-typedef std::vector<Byte>               SDR_dense_t;
-typedef std::vector<UInt>               SDR_sparse_t;
-typedef std::vector<std::vector<UInt>>  SDR_coordinate_t;
-typedef std::function<void()>           SDR_callback_t;
+using ElemDense        = Byte; //TODO allow changing this
+using ElemSparse       = UInt32; //must match with connections::CellIdx 
```

Comment:
ElemSparse & ElemDense represent data type of elements in SDR_sparse/dense_t; 
- ElemSparse is used to match in Connections with CellIdx
- ElemDense is in preparation for changing this (at runtime with teplate?)